src/platform/datapath_raw_socket.c

```diff
@@ -371,28 +370,72 @@ CxPlatResolveRoute(
     //
     // Map the next hop IP address to a link-layer address.
     //
-    MIB_IPNET_ROW2 IpnetRow = {0};
-    IpnetRow.InterfaceLuid = IpforwardRow.InterfaceLuid;
+    IpnetRow->InterfaceLuid = IpforwardRow.InterfaceLuid;
     if (QuicAddrIsWildCard(&IpforwardRow.NextHop)) { // On-link?
-        IpnetRow.Address = Route->RemoteAddress;
+        IpnetRow->Address = Route->RemoteAddress;
     } else {
-        IpnetRow.Address = IpforwardRow.NextHop;
+        IpnetRow->Address = IpforwardRow.NextHop;
     }
 
     //
-    // First call GetIpNetEntry2 to see if there's already a cached neighbor. If there
-    // isn't one, or if the cached neighbor's state is unreachable (which, NB, can happen
-    // in the case where a route lookup resulted in a dummy neighbor entry being created
-    // in TCPIP.sys) or incomplete, then queue up a solicitation event.
+    // Call GetIpNetEntry2 to see if there's already a cached neighbor.
     //
-    Status = GetIpNetEntry2(&IpnetRow);
+    Status = GetIpNetEntry2(IpnetRow);
+    if (Status != ERROR_SUCCESS || IpnetRow->State <= NlnsIncomplete) {
+        Status = ERROR_IO_PENDING;
```

Comment:
This is an unidiomatic use of PENDING (because there's no actual pending operation that has been kicked off) and confused me when I was reading CxPlatRouteResolutionWorkerThread- please change

Reply:
how about QUIC_STATUS_CONTINUE (ERROR_CONTINUE)?

Reply:
That sounds better, but still not immediately obvious, so let's add a code comment about what the status indicates to the caller.

Reply:
We can't use ERROR_CONTINUE unless we want that if (Status > 0) logic because ERROR_CONTINUE is positive too like PENDING.

Reply:
I think NOT_FOUND is better. It means next hop not found